crates/extricrate/src/lib.rs

```diff
@@ -3,19 +3,28 @@
 pub mod dependencies {
     use std::collections::HashMap;
 
+    #[derive(Debug, PartialEq, Eq)]
     pub struct ModuleName(String);
+    impl ModuleName {
+        fn new(name: &str) -> Self {
```

Comment:
It's better to take a String, since you'll need it anyway. That way, the caller can give you one they don't need anymore, preventing a copy.

Reply:
I was thinking that the flexibility of the api might beat an extra allocation, but happy to follow your lead 👍